### ISSUE TITLE
test_int64_overflow failure

### ISSUE THREAD

**Issue by toniatop** [pydata/pandas#11332]
Testing pandas 0.17:
```python
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-57-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_AU.UTF-8

pandas: 0.17.0
nose: 1.3.7
pip: 7.1.2
setuptools: 17.1.1
Cython: 0.22
numpy: 1.10.1
scipy: 0.16.0
statsmodels: 0.6.1
IPython: 4.0.0
sphinx: None
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.6
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.4.4
matplotlib: 1.4.3
openpyxl: 2.2.3
xlrd: 0.9.3
xlwt: 1.0.0
xlsxwriter: None
lxml: None
bs4: 4.3.2
html5lib: 1.0b6
httplib2: 0.9.1
apiclient: 1.4.0
sqlalchemy: 1.0.5
pymysql: None
psycopg2: None
```

nosetests pandas fail at:
`test_int64_overflow (pandas.io.tests.test_parsers.TestCParserHighMemory) ... Segmentation fault (core dumped)`

**Comment by jreback** [comment]
well since this passes linux-64 on all tests systems, I would suspect that its maybe the LANG binding (just a guess)

can you show

```
np.iinfo(np.int64).max
np.iinfo(np.int64).min
```

also putting a break point inside that test and showing exactly where it fails would be helpful

**Comment by toniatop** [comment]
The test fails at the very beginning:
`result = self.read_csv(StringIO(data))`

I can also reproduce the error in a interactive session:
```python
data = """ID
00013007854817840016671868
00013007854817840016749251
00013007854817840016754630
00013007854817840016781876
00013007854817840017028824
00013007854817840017963235
00013007854817840018860166"""
import pandas as pd
from StringIO import StringIO
result = pd.read_csv(StringIO(data))
```

**Comment by evanpw** [comment]
This is a packaging problem. The copy of parser.c (generated by cython) in the pip package is inconsistent with the version of parser.pyx that it's supposed to be generated from, and the packaged version has this bug on integer overflow.

In particular, notice that parser.c has a function called _try_int64_internal, while parser.pyx does not.

**Comment by jreback** [comment]
right this is the same as #11374 and fixed for 0.17.1
